docs/architecture/adr-102-rpc-companion.md

```diff
@@ -0,0 +1,341 @@
+expected that integrators and operators propose and design their own solutions based on their specific use-cases.
+
+This ADR provides a reference implementation that can be used and adapted for individual use-cases.
+
+## Decision
+
+TBD
+
+## Detailed Design
+
+### Requirements
+
+The target audience for this solution are operators and integrators that want to alleviate the load on their nodes by offloading
+the queryable data requests to the **RPC Companion**.
+
+This solution shall meet the following requirements in order to provide real benefits to these users.
+
+The **RPC Companion** solution shall:
+
+1. Provide an **[Ingest Service](#ingest-service)** implemented as a data companion that can pull data from the node and store it on
+its own storage (database)
+2. Provide its own storage ([Database](#database)) that can handle a high-load of reads and also have a good performance on inserting data.
+3. Implement a [database schema](#database-schema) that needs to be backwards compatible with the current CometBFT RPC.
+4. Do not enforce any breaking changes to the existing RPC.
+5. Ensure the responses returned by the [RPC Companion v1 endpoint](#rpc-endpoint) is wire compatible with the existing CometBFT endpoint.
+6. Implement tests to verify backwards compatibility.
+7. Be able to handle multiple concurrent requests and return idempotent responses.
+8. Do not crash or panic if the querying demand is very high or large responses are returned.
+9. Storage optimized for an access pattern of faster read throughput than write.
+10. Provide metrics to allow operators to properly monitor the services and infrastructure.
+
+It is ***NOT*** in the scope of the **RPC Companion**:
+
+1. Provide an authentication mechanism for the RPC Companion endpoint.
+
+### [RPC Endpoint](#rpc-endpoint)
+
+The RPC Companion endpoint will be the same as the CometBFT endpoint but with a `/v1` appended to it. The RPC Companion endpoint
+might also use a different port than the default CometBFT RPC port (e.g. `26657`).
+
+For example, suppose these are the URLs for each RPC endpoint:
+
+CometRPC -> `http://cosmos.host:26657`
+
+RPC Companion -> `http://rpc-companion.host:8080/v1`
+
+To make a request for a `block` at height `5` using the CometBFT endpoint:
+
+`curl --header "Content-Type: application/json" --request POST --data '{"method": "block", "params": ["5"], "id": 1}' http://cosmos.host:26657`
+
+The make the same request to the RPC Companion endpoint:
+
+`curl --header "Content-Type: application/json" --request POST --data '{"method": "block", "params": ["5"], "id": 1}' http://rpc-companion.host:8080/v1`
+
+> Note that only the URL changes between these two `curl` commands
+
+The RPC Companion will accept JSONRPC requests, the same way as the CometBFT RPC endpoint does.
+
+The methods for the RPC Companion endpoint on the following table are the ones that can be implemented first since it seems it will be straightforward and less complex
```

Comment:
Did you build this table from scratch, or is it present elsewhere in the documentation? If the second case, it would be good to reference the original table (if one). In any case, it would be good to reference when the reader can find a full description of the methods.

Reply:
it was built from scratch. We do have an RPC documentation in the docs.